stripe/src/main/java/com/stripe/android/view/CardNumberEditText.java

```diff
@@ -0,0 +1,175 @@
+package com.stripe.android.view;
+
+import android.content.Context;
+import android.support.annotation.VisibleForTesting;
+import android.text.Editable;
+import android.text.InputFilter;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.widget.EditText;
+
+import com.stripe.android.model.Card;
+import com.stripe.android.util.StripeTextUtils;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * An {@link EditText} that handles spacing out the digits of a credit card.
+ */
+public class CardNumberEditText extends EditText {
+
+    private static final int MAX_LENGTH_COMMON = 19;
+    // Note that AmEx and Diners Club have the same length
+    // because Diners Club has one more space, but one less digit.
+    private static final int MAX_LENGTH_AMEX_DINERS = 17;
+
+    private static final Integer[] SPACES_ARRAY_COMMON = {4, 9, 14};
+    private static final Set<Integer> SPACE_SET_COMMON =
+            new HashSet<>(Arrays.asList(SPACES_ARRAY_COMMON));
+
+    private static final Integer[] SPACES_ARRAY_AMEX = {4, 11};
+    private static final Set<Integer> SPACE_SET_AMEX =
```

Comment:
YES! I forgot about that one. Was trying to avoid adding another library (like guava's `Sets.newHashSet`)